### ISSUE TITLE
Slint panics using backend-winit-wayland with ScrollView inside Layout

### ISSUE THREAD

**Issue by dirthsj** [slint-ui/slint#2099]
Summary: Running the example below in Wayland crashes immediately during launch. I have tried to create a more minimal example but was unable to. Switching `backend-winit-wayland` for `backend-winit-x11` does not trigger the panic.
Cargo.toml:

 `slint = { version = "0.3.3", default-features = false, features = ["renderer-winit-femtovg", "backend-winit-wayland", "compat-0-3-0"] }`

<details>
<summary>Slint Content</summary>

```
import { VerticalBox, HorizontalBox, ScrollView, Button } from "std-widgets.slint";
MainWindow := Window {
    property<string> chapter-text;
    
    HorizontalBox {
        VerticalBox {
            width: 150px;
            Button { text: "First"; height: 50px; }
            Button { text: "Second"; height: 50px; }
            Button { text: "Third"; height: 50px; }
        }
        ScrollView {
            preferred-height: 800px;
            min-width: 100px;
            min-height: 100px;
            viewport-height: max(self.visible-height, text.preferred-height);
            text := Text {
                width: parent.width;
                wrap: word-wrap;
                text: chapter-text;
            }
        }
    }
    min-height: 768px;
    min-width: 1366px;
    preferred-height: 1080px;
    preferred-width: 1920px;
}
```
</details>

Main function:
```rust
fn main() {
    MainWindow::new().run();
}
```

<details>
<summary>RUSTC_BACKTRACE=full</summary>

```
xdg_toplevel@24: error -1: width and height must be positive or zero
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Error { raw_code: Some(12289), raw_os_message: None, kind: InitializationFailed }', /home/dirthsj/.cargo/registry/src/github.com-1ecc6299db9ec823/i-slint-backend-winit-0.3.3/glcontext.rs:144:84
stack backtrace:
   0:     0x562de2c0c8b0 - std::backtrace_rs::backtrace::libunwind::trace::h1d00f3fcf4cb5ac4
                               at /rustc/90743e7298aca107ddaa0c202a4d3604e29bfeb6/library/std/src/../../backtrace/src/backtrace/libunwind.rs:93:5
   1:     0x562de2c0c8b0 - std::backtrace_rs::backtrace::trace_unsynchronized::h920a6ff332484ee2
                               at /rustc/90743e7298aca107ddaa0c202a4d3604e29bfeb6/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   2:     0x562de2c0c8b0 - std::sys_common::backtrace::_print_fmt::hd7323920c925af6d
                               at /rustc/90743e7298aca107ddaa0c202a4d3604e29bfeb6/library/std/src/sys_common/backtrace.rs:65:5
   3:     0x562de2c0c8b0 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h3155a8c966b4beb5
                               at /rustc/90743e7298aca107ddaa0c202a4d3604e29bfeb6/library/std/src/sys_common/backtrace.rs:44:22
   4:     0x562de2c3304e - core::fmt::write::h062c617411b691df
                               at /rustc/90743e7298aca107ddaa0c202a4d3604e29bfeb6/library/core/src/fmt/mod.rs:1209:17
   5:     0x562de2c06165 - std::io::Write::write_fmt::hb61fdf1275c61e1c
                               at /rustc/90743e7298aca107ddaa0c202a4d3604e29bfeb6/library/std/src/io/mod.rs:1682:15
   6:     0x562de2c0c675 - std::sys_common::backtrace::_print::hd1b4d9664ab500e0
                               at /rustc/90743e7298aca107ddaa0c202a4d3604e29bfeb6/library/std/src/sys_common/backtrace.rs:47:5
   7:     0x562de2c0c675 - std::sys_common::backtrace::print::hca896ae22beb06cb
                               at /rustc/90743e7298aca107ddaa0c202a4d3604e29bfeb6/library/std/src/sys_common/backtrace.rs:34:9
   8:     0x562de2c0df2f - std::panicking::default_hook::{{closure}}::h0b5eeed5cf36ab5f
                               at /rustc/90743e7298aca107ddaa0c202a4d3604e29bfeb6/library/std/src/panicking.rs:267:22
   9:     0x562de2c0dc6a - std::panicking::default_hook::h8932b573145a321b
                               at /rustc/90743e7298aca107ddaa0c202a4d3604e29bfeb6/library/std/src/panicking.rs:286:9
  10:     0x562de2c0e628 - std::panicking::rust_panic_with_hook::h4b1447a24e3e94f8
                               at /rustc/90743e7298aca107ddaa0c202a4d3604e29bfeb6/library/std/src/panicking.rs:688:13
  11:     0x562de2c0e3c7 - std::panicking::begin_panic_handler::{{closure}}::h8701da9995a3820c
                               at /rustc/90743e7298aca107ddaa0c202a4d3604e29bfeb6/library/std/src/panicking.rs:579:13
  12:     0x562de2c0cd5c - std::sys_common::backtrace::__rust_end_short_backtrace::hb696c5ed02a01598
                               at /rustc/90743e7298aca107ddaa0c202a4d3604e29bfeb6/library/std/src/sys_common/backtrace.rs:137:18
  13:     0x562de2c0e0e2 - rust_begin_unwind
                               at /rustc/90743e7298aca107ddaa0c202a4d3604e29bfeb6/library/std/src/panicking.rs:575:5
  14:     0x562de25dd773 - core::panicking::panic_fmt::h8aa706a976963c88
                               at /rustc/90743e7298aca107ddaa0c202a4d3604e29bfeb6/library/core/src/panicking.rs:65:14
  15:     0x562de25dda33 - core::result::unwrap_failed::h065c02f906ca4578
                               at /rustc/90743e7298aca107ddaa0c202a4d3604e29bfeb6/library/core/src/result.rs:1791:5
  16:     0x562de26c421b - core::result::Result<T,E>::unwrap::h702001ae40306bca
                               at /rustc/90743e7298aca107ddaa0c202a4d3604e29bfeb6/library/core/src/result.rs:1113:23
  17:     0x562de26c421b - i_slint_backend_winit::glcontext::OpenGLContext::new_context::h50bac4b1fcd015e3
                               at /home/dirthsj/.cargo/registry/src/github.com-1ecc6299db9ec823/i-slint-backend-winit-0.3.3/glcontext.rs:144:17
  18:     0x562de26c9005 - <i_slint_backend_winit::renderer::femtovg::FemtoVGRenderer as i_slint_backend_winit::renderer::WinitCompatibleRenderer>::show::h08ba92d30774d0c5
                               at /home/dirthsj/.cargo/registry/src/github.com-1ecc6299db9ec823/i-slint-backend-winit-0.3.3/renderer/femtovg.rs:57:30
  19:     0x562de26957c7 - <i_slint_backend_winit::glwindow::GLWindow<Renderer> as i_slint_core::window::WindowAdapterSealed>::show::{{closure}}::h8f2074a00c7a62cc
                               at /home/dirthsj/.cargo/registry/src/github.com-1ecc6299db9ec823/i-slint-backend-winit-0.3.3/glwindow.rs:540:13
  20:     0x562de26957c7 - core::ops::function::FnOnce::call_once::h822d1fd557d12edf
                               at /rustc/90743e7298aca107ddaa0c202a4d3604e29bfeb6/library/core/src/ops/function.rs:251:5
  21:     0x562de26957c7 - i_slint_backend_winit::glwindow::GLWindow<Renderer>::call_with_event_loop::hd9a6bbbadd90f6c7
                               at /home/dirthsj/.cargo/registry/src/github.com-1ecc6299db9ec823/i-slint-backend-winit-0.3.3/glwindow.rs:175:9
  22:     0x562de26957c7 - <i_slint_backend_winit::glwindow::GLWindow<Renderer> as i_slint_core::window::WindowAdapterSealed>::show::h19b8195a61c68999
                               at /home/dirthsj/.cargo/registry/src/github.com-1ecc6299db9ec823/i-slint-backend-winit-0.3.3/glwindow.rs:385:9
  23:     0x562de2ab1ec9 - i_slint_core::window::WindowInner::show::hac6a411d6c759cdc
                               at /home/dirthsj/.cargo/registry/src/github.com-1ecc6299db9ec823/i-slint-core-0.3.3/window.rs:609:9
  24:     0x562de25f0a1e - <untitled::slint_generatedMainWindow::MainWindow as i_slint_core::api::ComponentHandle>::show::hbf0d4348e8d7c87b
                               at /home/dirthsj/CLionProjects/untitled/src/main.rs:10:1
  25:     0x562de25f09e3 - <untitled::slint_generatedMainWindow::MainWindow as i_slint_core::api::ComponentHandle>::run::hfecae0cdad1fa1b4
                               at /home/dirthsj/CLionProjects/untitled/src/main.rs:10:1
  26:     0x562de2619c5d - untitled::main::h23fd18a4f502a139
                               at /home/dirthsj/CLionProjects/untitled/src/main.rs:104:5
  27:     0x562de260df4b - core::ops::function::FnOnce::call_once::hd477007b990c24d8
                               at /rustc/90743e7298aca107ddaa0c202a4d3604e29bfeb6/library/core/src/ops/function.rs:251:5
  28:     0x562de263abde - std::sys_common::backtrace::__rust_begin_short_backtrace::he823076b4dded99f
                               at /rustc/90743e7298aca107ddaa0c202a4d3604e29bfeb6/library/std/src/sys_common/backtrace.rs:121:18
  29:     0x562de26199d1 - std::rt::lang_start::{{closure}}::hfaa8db6c98182ab0
                               at /rustc/90743e7298aca107ddaa0c202a4d3604e29bfeb6/library/std/src/rt.rs:166:18
  30:     0x562de2c0173b - core::ops::function::impls::<impl core::ops::function::FnOnce<A> for &F>::call_once::h8cbb48ae40ddb046
                               at /rustc/90743e7298aca107ddaa0c202a4d3604e29bfeb6/library/core/src/ops/function.rs:286:13
  31:     0x562de2c0173b - std::panicking::try::do_call::h92db802eb38b49b7
                               at /rustc/90743e7298aca107ddaa0c202a4d3604e29bfeb6/library/std/src/panicking.rs:483:40
  32:     0x562de2c0173b - std::panicking::try::ha8949d2082cf3644
                               at /rustc/90743e7298aca107ddaa0c202a4d3604e29bfeb6/library/std/src/panicking.rs:447:19
  33:     0x562de2c0173b - std::panic::catch_unwind::h5e34c1f8a5992ed9
                               at /rustc/90743e7298aca107ddaa0c202a4d3604e29bfeb6/library/std/src/panic.rs:137:14
  34:     0x562de2c0173b - std::rt::lang_start_internal::{{closure}}::hea52a0bb3f8ff16a
                               at /rustc/90743e7298aca107ddaa0c202a4d3604e29bfeb6/library/std/src/rt.rs:148:48
  35:     0x562de2c0173b - std::panicking::try::do_call::h5bc358faf3d68a8b
                               at /rustc/90743e7298aca107ddaa0c202a4d3604e29bfeb6/library/std/src/panicking.rs:483:40
  36:     0x562de2c0173b - std::panicking::try::h675304212928379d
                               at /rustc/90743e7298aca107ddaa0c202a4d3604e29bfeb6/library/std/src/panicking.rs:447:19
  37:     0x562de2c0173b - std::panic::catch_unwind::h7ce3ad349ed5c844
                               at /rustc/90743e7298aca107ddaa0c202a4d3604e29bfeb6/library/std/src/panic.rs:137:14
  38:     0x562de2c0173b - std::rt::lang_start_internal::hcd7e45acd25ab5ab
                               at /rustc/90743e7298aca107ddaa0c202a4d3604e29bfeb6/library/std/src/rt.rs:148:20
  39:     0x562de26199aa - std::rt::lang_start::h3cc2f66b5fac5820
                               at /rustc/90743e7298aca107ddaa0c202a4d3604e29bfeb6/library/std/src/rt.rs:165:17
  40:     0x562de2619cde - main
  41:     0x7f9482f6a510 - __libc_start_call_main
  42:     0x7f9482f6a5c9 - __libc_start_main@GLIBC_2.2.5
  43:     0x562de25ddc15 - _start
  44:                0x0 - <unknown>
```
</details>

System details:
- Fedora Linux 37
- KDE Plasma 5.26.5
- KDE Frameworks 5.102.0
- Kernel Version: 6.1.6-200.fc37.x86_64
- Graphics Platform: Wayland
- Graphics Processor: NVIDIA GeForce RTX 3060 Ti

**Comment by tronical** [comment]
Thank you for your very detailed report! I haven't tried to reproduce it yet, but it all sounds very plausible - the order of events between wayland and x11 are different and the panic indicates that somehow the width and height of the surface is somehow negative.

I have one question though: What is the resolution of your screen?

**Comment by dirthsj** [comment]
I have 2x 1440p displays.

**Comment by dirthsj** [comment]
After updating to version `0.3.5` I can no longer reproduce this bug. Closing.

**Comment by tronical** [comment]
Thanks for checking :)
